doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+is given with enough of the expected error message attached to the exception
+to ensure that the expected exception is raised.
+
+For example, the function `scipy.stats.zmap` is supposed to raise a
+``ValueError`` if the input contains ``nan`` and ``nan_policy`` is ``"raise"``.
+A test for this is::
+
+    scores = np.array([1, 2, 3])
+    compare = np.array([-8, -3, 2, 7, 12, np.nan])
+    with pytest.raises(ValueError, match='input contains nan'):
+        stats.zmap(scores, compare, nan_policy='raise')
+
+The ``match`` argument ensures that the test doesn't pass by raising
+a ``ValueError`` that is not related to the input containing ``nan``.
+
+
+A parameter called lambda
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are functions where convention dictates that (ideally) a parameter
+should be called 'lambda'.  In Python, 'lambda' is a reserved word that
+cannot be used as a variable name.  If a new function or class has a
+parameter where a name that is similar to 'lambda' is really the best
+choice, the preferred name to use is [TBD: ``lam``, ``lmb``, ``lamb``,
+``lmbda``, ``lambda_``?].
+
+There has been a lack of consistency in the past:
+
+============================== =============
+Function or class              Variable name
+============================== =============
+`scipy.stats.boxcox`           ``lmbda``
+`scipy.stats.boxcox_llf`       ``lmb``
+`scipy.stats.yeojohnson`       ``lmbda``
+`scipy.stats.yeojohnson_llf`   ``lmb``
+`scipy.stats.chi2_contingency` ``lambda_``
+`scipy.stats.boltzmann`        ``lambda_``
+`scipy.stats.planck`           ``lambda_``
+`scipy.stats.tukeylambda`      ``lam``
+`scipy.signal.cspline1d`       ``lamb``
+`scipy.signal.qspline1d`       ``lamb``
+`scipy.signal.cspline2d`       ``lambda`` [*]
+`scipy.signal.qspline2d`       ``lambda`` [*]
+`scipy.signal.spline_filter`   ``lmbda``
+`scipy.special.tklmbda`        ``lmbda``
+`scipy.special.boxcox`         ``lmbda``
+`scipy.special.boxcox1p`       ``lmbda``
+`scipy.special.inv_boxcox`     ``lmbda``
+`scipy.special.inv_boxcox1p`   ``lmbda``
+============================== =============
+
+.. [*] This function accepts positional arguments only, but the docstring
+       documents the argument as `lambda`.
+
+
+Implicit string continuation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When an implicit continuation is used to create a long literal string, the
+string is generally split at a space, and the space is included in the
+second part of the string.
+
+  Yes::
+
+      long_string = ("This is a very long string that demonstrates"
+                     " this guideline.")
+
+  No::
+
+      long_string = ("This is a very long string that demonstrates "
+                     "this guideline.")
+
+
+Documentation Guidelines
+------------------------
+
+Use of LaTeX
+~~~~~~~~~~~~
+[TBD.  In the past, we had an unwritten rule that LaTex markup should be
+restricted to the Notes section.  We don't follow that rule these days.
+Should we?]
```

Comment:
How about something along the lines of


When writing out formulas, consider using pseudocode or TeX markup. The former is preferred for simple short formulas but gets unwieldy for more complicated ones. In the latter case, you may use TeX markup instead. When using TeX markup, please consider TeX markup as souce code --- hence aim for readability. 
These are not hard and fast rules, more of general guidelines: 

- keep the number of curly braces to a minimum.
- avoid excessive markup control, e.g. `\left( `,  `\bigl` and so on.
- prefer simple, more readable constructs. For instance, consider using `/` instead of `\frac{...}{...}`

For instance

```
No:   $\frac{1}{\sqrt{2\pi\, \sigma^{2}}} e^{-\bigl(\frac{x - \mu}{2\sigma}\bigr)^2}$

Yes: $\exp(-(x-\mu)^2 / \sigma^2) / \sqrt(2\pi\sigma^2) $
```

Reply:
Even though I won't recommend that SciPy use it, there is a technical solution to the "LaTeX issue".  For my [`mpsci`](https://github.com/WarrenWeckesser/mpsci) repository, I created a Sphinx extension that uses regular expressions to replace text in a docstring when Sphinx is run.  For example, the text of the docstring for ``mpsci.fun.marcumq`` is

https://github.com/WarrenWeckesser/mpsci/blob/f524ed8ab3a4c84485e253b42c5c82c1030ab248/mpsci/fun/_marcumq.py#L14-L43

which is

```
def marcumq(m, a, b):
    """
    The Marcum Q function.

    The function is defined as the integral from b to infinity of

        x*(x/a)**(M-1)*exp(-(x**2 + a**2)/2)*I_{M-1}(a*x)

    where I_{M-1}(x) is the modified Bessel function of order M-1.

    *See also:* `mpsci.fun.cmarcumq`

    Returns
    -------
    q : mpmath.mpf
        The function value.

    Notes
    -----
    The function uses numerical integration, so it can be very slow.

    Examples
    --------
    >>> import mpmath
    >>> from mpsci.fun import marcumq
    >>> mpmath.mp.dps = 40
    >>> marcumq(2, 0.5, 3.0)
    mpf('0.07558718754263240906718640000640082605610073')

    """
```
There is no explicit LaTeX markup. (I guess  `I_{M-1}(x)` is LaTeXish, but it is not wrapped with the Sphinx `:math:` markup.) The HTML output, however, is [here](https://warrenweckesser.github.io/mpsci/fun/marcumq.html#mpsci.fun.marcumq), with integral signs, fractions, big parentheses, etc.

The reason I don't recommend it for SciPy is that maintaining the regular expressions is a fussy and fragile process.  It is easy to make a few changes to a docstring and forget to also update the regular expression that is used to match the text that is to be replaced with LaTeX markup.  I don't mind doing it in my one-man project, but I think it would be too much extra maintenance for SciPy.  I point it out just so folks can see what is *possible* (and maybe see if anyone can come up with a more robust method to do it).

Reply:
I'm removing the section about the use of LaTeX from this PR.  If there is interest in continuing the discussion, it can be done in a follow-up PR.